src/modules/dns.c

```diff
@@ -395,6 +395,11 @@ static int dns_module_init(noit_module_t *self) {
 
   conf = noit_module_get_userdata(self);
 
+  mtev_hash_init(&dns_rtypes);
```

Comment:
This is great but it won't fix the bug.. we need to call `mtev_hash_init_locks` everywhere and choose `MTEV_HASH_LOCK_MODE_MUTEX`